examples/src/main/java/com/mapbox/navigation/examples/ui/BuildingExtrusionActivity.kt

```diff
@@ -0,0 +1,168 @@
+    private var colorList = listOf(Color.BLUE, Color.MAGENTA, Color.parseColor("#32a88f"))
+    private var opacityList = listOf(.5f, .2f, .8f)
+    private var adjustExtrusionsStyleButtonIndex = 0
+    private val route by lazy { getDirectionsRoute() }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_building_extrusion)
+
+        navigationView.onCreate(savedInstanceState)
+        navigationView.initialize(this, getInitialCameraPosition())
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        navigationView.onLowMemory()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        navigationView.onStart()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        navigationView.onResume()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        navigationView.onStop()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        navigationView.onPause()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        navigationView.onDestroy()
+    }
+
+    override fun onBackPressed() {
+        // If the navigation view didn't need to do anything, call super
+        if (!navigationView.onBackPressed()) {
+            super.onBackPressed()
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        navigationView.onSaveInstanceState(outState)
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+        navigationView.onRestoreInstanceState(savedInstanceState)
+    }
+
+    override fun onNavigationReady(isRunning: Boolean) {
+        if (!isRunning && !::navigationMapboxMap.isInitialized) {
+            ifNonNull(navigationView.retrieveNavigationMapboxMap()) { navMapboxMap ->
+                this.navigationMapboxMap = navMapboxMap
+                this.navigationMapboxMap.updateLocationLayerRenderMode(RenderMode.NORMAL)
+                this.mapboxMap = navMapboxMap.retrieveMap()
+
+                navigationView.retrieveMapboxNavigation()?.let { this.mapboxNavigation = it }
+
+                val optionsBuilder = NavigationViewOptions.builder()
+                optionsBuilder.navigationListener(this)
+                optionsBuilder.directionsRoute(route)
+                optionsBuilder.shouldSimulateRoute(true)
+                optionsBuilder.routeProgressObserver(this)
+                optionsBuilder.navigationOptions(NavigationOptions.Builder().build())
+                navigationView.startNavigation(optionsBuilder.build())
+
+                // Initialize the Navigation UI SDK's BuildingExtrusionLayer class.
+                buildingExtrusionLayer = BuildingExtrusionLayer(mapboxMap,
+                        navigationView.findViewById(R.id.navigationMapView))
```

Comment:
This might not be the best developer experience. If we're moving the layer initialization out of the Nav UI SDK, we should also offer a clean way to get the `MapView`'s reference instead of forcing a developer to find it by ID, which could be hard to discover and error-prone.

Reply:
As discussed offline, `MapView` is no longer needed by this layer.